test/addons/make-callback-recurse/test.js

```diff
@@ -132,38 +132,20 @@ function checkDomains() {
     }));
   }), 1);
 
-  // Make sure nextTick, setImmediate and setTimeout can all recover properly
-  // after a thrown makeCallback call.
-  process.nextTick(common.mustCall(function() {
+  function testTimer(id) {
+    // Make sure nextTick, setImmediate and setTimeout can all recover properly
+    // after a thrown makeCallback call.
     const d = domain.create();
     d.on('error', common.mustCall(function(e) {
-      assert.strictEqual(e.message, 'throw from domain 3');
+      assert.strictEqual(e.message, `throw from domain ${id}`);
     }));
     makeCallback({domain: d}, function() {
-      throw new Error('throw from domain 3');
+      throw new Error(`throw from domain ${id}`);
     });
     throw new Error('UNREACHABLE');
-  }));
+  }
 
-  setImmediate(common.mustCall(function() {
-    const d = domain.create();
-    d.on('error', common.mustCall(function(e) {
-      assert.strictEqual(e.message, 'throw from domain 2');
-    }));
-    makeCallback({domain: d}, function() {
-      throw new Error('throw from domain 2');
-    });
-    throw new Error('UNREACHABLE');
-  }));
-
-  setTimeout(common.mustCall(function() {
-    const d = domain.create();
-    d.on('error', common.mustCall(function(e) {
-      assert.strictEqual(e.message, 'throw from domain 1');
-    }));
-    makeCallback({domain: d}, function() {
-      throw new Error('throw from domain 1');
-    });
-    throw new Error('UNREACHABLE');
-  }));
+  process.nextTick(common.mustCall(testTimer.bind(null, 3)));
```

Comment:
Because it's late and I'm tired and not thinking as clearly as I should be.

Oh, wait, it's 8:41 PM. Can't use that excuse.

Anyway, yes, changed! Thanks.